mscore/seq.cpp

```diff
@@ -1405,10 +1415,13 @@ void Seq::putEvent(const NPlayEvent& event, unsigned framePos)
             qDebug("bad channel value %d >= %d", channel, cs->midiMapping()->size());
             return;
             }
+
+      // dispatch midi messages to MIDI driver first before rendering with MuseScore's synth, in order to avoid additional unpredictable latency associated with MuseScore's synth
+      if (_driver != 0 && (preferences.useJackMidi || preferences.useAlsaAudio || preferences.usePortaudioAudio))
+            _driver->putEvent(event, framePos);
```

Comment:
The more I think about it, I believe there is a performance bug in the actual processing of a large number of notes at a particular tick and not necessarily the rendering of it.  It might have something to do with cache usage.  The interesting thing is that Seq::putEvent is only handling one event at a time, so my change only reordered order of midi dispatch and rendering of each event individually, but it doesn't reorder all midi dispatch for the segment to occur before the rendering of all notes of that segment.  I'm going to look into this more...